### PR TITLE
Choose layers from URL

### PR DESCRIPTION
**Pull request by danielkryska**
- The add wms form should reload layers on a URL changes (except layers changes)
- User should be able to choose layers by it's titles
- User should be able to use additional query params accepted by Geo-server
- Tiles addition should have 2 step validation 
  - A wms capabilities validation while loading metadata
  - The Wms image validation while saving
- When any layers is added in the URL, checkboxes contained layer should be checked
- When a URL don't contain LAYERS= query param, all layers from metadata are added into the link

image

image

### CODE REVIEW

**Comment by Marcelinna** [comment]
"URL powinien mieć wybraną conajmniej 1 warstwę" -> "co najmniej" is correctly

**Comment by Marcelinna** [comment]
When user change url (delete previous and write new), see error "Wystąpił błąd i może być on związany z Cross-Origin Request Blocked: Polityka administracyjna serwera nie zezwala na czytanie przez źródła zewnętrzne. Błąd: TypeError: t.getAttribute(...) is null" without having to click on the button <Dodaj>

**Comment by Marcelinna** [comment]
After click on the button <_Dodaj> and turn on layer to the map, the added layer is duplicated and all the others previously removed are added too.

File: s4e-web/src/app/views/map-view/view-manager/view-manager.component.ts [hunk 1]
```diff
@@ -112,6 +112,7 @@ export class ViewManagerComponent implements OnInit, OnDestroy {
   }
 
   selectOverlay(overlayId: number) {
+    console.log(overlayId, this.overlayQuery.getAll());
```
Comment (jswk) [inline s4e-web/src/app/views/map-view/view-manager/view-manager.component.ts:115]: A stray console.log

**Review by jswk — APPROVED** [review]
I didn't notice any malfunction, the PR seems fine.
Just clean the remaining console.log and it's ready to go.

**Comment by Marcelinna** [comment]
When user type URL and after click button <_Anuluj> every layer are still visible 
image

**Comment by Marcelinna** [comment]
if the user enters the URL but gets an error, changing the URL does not reload the layers either and after click the button <_Dodaj> sees the error
image

**Comment by Marcelinna** [comment]
When I enter https://wody.isok.gov.pl/gpservices/KZGW/ISOK_MPHP/MapServer/WMSServer?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20 error "Odpowiedz nie jest zdjeciem" 
image

**Comment by Marcelinna** [comment]
https://wody.isok.gov.pl/gpservices/KZGW/MZP20_Predkosc_WysokiePrawdopodPowodzi/MapServer/WMSServer?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=1,2,3,4,5,6,7,8,9,10

image

**Comment by danielkryska** [comment]
> When user type URL and after click button <_Anuluj> and return to adding new wms, every layer are still visible, and when type new URL, the layer doesnt reloaded
> image

done

**Comment by danielkryska** [comment]
> When I enter https://wody.isok.gov.pl/gpservices/KZGW/ISOK_MPHP/MapServer/WMSServer?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20 error "Odpowiedz nie jest zdjeciem" is displayed. Please check this URL (on master I can add it correctly)
> 
> image

The geoserver URL has a specific way of throwing CORS I belive. 

Form is sending request for
```
https://wody.isok.gov.pl/gpservices/KZGW/ISOK_MPHP/MapServer/WMSServer?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities
```

On localhost the response is `400 bad request`, but when you type the get capabilities URL response is valid XML.

**Comment by danielkryska** [comment]
> https://wody.isok.gov.pl/gpservices/KZGW/MZP20_Predkosc_WysokiePrawdopodPowodzi/MapServer/WMSServer?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=1,2,3,4,5,6,7,8,9,10
> 
> image

**Done**

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**